cmd/admin-handlers.go

```diff
@@ -80,3 +80,77 @@ func (adminAPI adminAPIHandlers) ServiceRestartHandler(w http.ResponseWriter, r
 	w.WriteHeader(http.StatusOK)
 	sendServiceCmd(globalAdminPeers, serviceRestart)
 }
+
+// ListLocksHandler - GET /?lock&bucket=mybucket&prefix=myprefix&older-than=rel_time
+// HTTP header x-minio-operation: list
+// ---------
+// Lists locks held on a given bucket, prefix and relative time.
+func (adminAPI adminAPIHandlers) ListLocksHandler(w http.ResponseWriter, r *http.Request) {
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, r, adminAPIErr, r.URL.Path)
+		return
+	}
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	prefix := vars["prefix"]
+	relTimeStr := vars["older-than"]
+
+	if !IsValidBucketName(bucket) {
+		writeErrorResponse(w, r, ErrInvalidBucketName, r.URL.Path)
+		return
+	}
+	if !IsValidObjectPrefix(prefix) {
+		writeErrorResponse(w, r, ErrInvalidObjectName, r.URL.Path)
+	}
+	relTime, err := time.ParseDuration(relTimeStr)
+	if err != nil {
+		writeErrorResponse(w, r, ErrInvalidDuration, r.URL.Path)
+		return
+	}
+
+	volLocks := filterLocks(bucket, prefix, relTime)
+	jsonBytes, err := json.Marshal(volLocks)
+	if err != nil {
+		writeErrorResponseNoHeader(w, r, ErrInternalError, r.URL.Path)
+		errorIf(err, "Failed to marshal lock information into json.")
+		return
+	}
+	writeSuccessResponse(w, jsonBytes)
+}
+
+// ClearLocksHandler - POST /?lock&bucket=mybucket&object=myobject
+// HTTP header x-minio-operation: clear
+// ---------
+// Clear locks held on a given bucket, object and relative time.
+func (adminAPI adminAPIHandlers) ClearLocksHandler(w http.ResponseWriter, r *http.Request) {
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, r, adminAPIErr, r.URL.Path)
+		return
+	}
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	object := vars["object"]
+
+	if !IsValidBucketName(bucket) {
+		writeErrorResponse(w, r, ErrInvalidBucketName, r.URL.Path)
+		return
+	}
+	if !IsValidObjectName(object) {
+		writeErrorResponse(w, r, ErrInvalidObjectName, r.URL.Path)
+	}
+
+	// Fetch lock information on bucket/object that are available
+	// now (i.e, relTime is 0s in the past).
+	volLocks := filterLocks(bucket, object, time.Duration(0))
+	jsonBytes, err := json.Marshal(volLocks)
+	if err != nil {
+		writeErrorResponseNoHeader(w, r, ErrInternalError, r.URL.Path)
+		errorIf(err, "Failed to marshal lock information into json.")
+		return
+	}
+	// Remove lock matching bucket/object
+	globalNSMutex.ForceUnlock(bucket, object)
+	writeSuccessResponse(w, jsonBytes)
```

Comment:
It returns the list of locks that belong to bucket, object that is supplied as query values. We use 0s as relative time to select all locks held on bucket, object.